packages/eslint-mdx/src/helpers.ts

```diff
@@ -90,8 +90,14 @@ export const loadModule = async <T>(modulePath: string): Promise<T> => {
         // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-unsafe-return
         return require(modulePath)
       } catch (err) {
+        const code = (err as { code: string }).code
         /* istanbul ignore if */
-        if ((err as { code: string }).code === 'ERR_REQUIRE_ESM') {
+        if (
+          code === 'ERR_REQUIRE_ESM' ||
+          // A pure ESM could have no `exports.require` and then throw the following error,
+          // related to #427.
+          code === 'ERR_PACKAGE_PATH_NOT_EXPORTED'
```

Comment:
Can‘t you use `import-meta-resolve`? That everything from conditions to import/export maps, etc.

Reply:
@wooorm `import-meta-resolve` is pure ESM only, `eslint-mdx` is commonjs, I don't understand how will it help? Any advice?

Reply:
> `eslint-mdx` is commonjs

I forgot that this project is still CJS.
Still, you could bundle `import-meta-resolve` (`npx esbuild import-meta-resolve --bundle --platform=node --outfile=vendor/import-meta-resolve.js`), or if the wrapping code is async, use a dynamic import expression to get it (`await import('xxx')`).

>  I don't understand how will it help? Any advice?

`import-meta-resolve` is 1500 lines (sloc). That’s a lot of code. It matches how Node imports ESM and CJS. The code here is about 50 lines, so it’s missing some things.

This PR is called “handle pure ESM robustly”, if you want to handle ESM/CJS really robustly, like Node.js does, then I recommend `import-meta-resolve`!

Reply:
This PR handlers pure ESM remark plugins more robustly, not itself, sorry it could be confusing for you.

Reply:
You are adding an exception message for export maps.
Your code here does not handle export maps.
`import-meta-resolve` handles export maps.

Reply:
`loadEsmModule` is using `import()` to load modules.

```ts
// Load the ESM configuration file using the TypeScript dynamic import workaround.
// Once TypeScript provides support for keeping the dynamic import this workaround can be
// changed to a direct dynamic import.
```

I'll change this project by using `node16` resolution, but that would a minor change.

Reply:
@wooorm I merged this PR for not blocking a patch release, I'll work on a minor change to use node16 resolution and native dynamic import later.